src/cljam/io/vcf/reader.clj

```diff
@@ -178,3 +182,46 @@
                     :deep (vcf-util/variant-parser (.meta-info rdr) (.header rdr))
                     :vcf identity)]
      (map parse-fn (read-data-lines (.reader rdr) (.header rdr) kws)))))
+
+(defn read-variants-randomly
+  [^VCFReader rdr {:keys [chr start end depth] :or {depth :deep start 1 end 4294967296}}]
+  (let [kws (mapv keyword (drop 8 (.header rdr)))
+        tabix-path (str (.getPath ^java.net.URL (.url rdr)) ".tbi")
+        tabix-data (tabix/read-index tabix-path)
```

Comment:
We should store delayed `tabix-data` in `VCFReader` for speeding up multiple querying.